### PR TITLE
CF + API keys + UsagePlan ~ security adjustments

### PR DESCRIPTION
**Pull request by sebolabs**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -28,6 +28,16 @@ variable "with_cloudfront" {
   description = "CloudFront on/off switch"
 }
 
+variable "wab_acl_id" {
+  type        = "string"
+  description = "WAF Web ACL id"
+}
+
+variable "x_api_key" {
```
Comment (przemos) [inline variables.tf]: please, be more descriptive like cf-apig-channel-key

File: variables.tf [hunk 2]
```diff
@@ -28,6 +28,16 @@ variable "with_cloudfront" {
   description = "CloudFront on/off switch"
 }
 
+variable "wab_acl_id" {
```
Comment (przemos) [inline variables.tf]: waf

File: 01_frontend.tf [hunk 3]
```diff
@@ -133,10 +133,11 @@ resource "aws_api_gateway_rest_api" "MotrWeb" {
...
 resource "aws_api_gateway_method" "LambdaRootGET" {
-  rest_api_id   = "${aws_api_gateway_rest_api.MotrWeb.id}"
-  resource_id   = "${aws_api_gateway_rest_api.MotrWeb.root_resource_id}"
-  http_method   = "GET"
-  authorization = "NONE"
+  rest_api_id      = "${aws_api_gateway_rest_api.MotrWeb.id}"
+  resource_id      = "${aws_api_gateway_rest_api.MotrWeb.root_resource_id}"
+  http_method      = "GET"
+  authorization    = "NONE"
+  api_key_required = "${var.with_cloudfront ? true : false}"
```
Comment (przemos) [inline 01_frontend.tf]: cant var.with_cloudfront be just true or false without ternary op ?

Comment (sebolabs) [inline 01_frontend.tf]: it can but it is more visible what value will be assigned and not what value is hidden behind the with_cloudfront variable

**Pull request merged**